### PR TITLE
Update Mixer-TTS, FastPitch and TTSDataset

### PR DESCRIPTION
**Pull request by Oktai15**
- [x] Update nemo.collections.tts.torch.data.TTSDataset and nemo.collections.tts.torch.data.MixerTTSDataset
    - [x] Rename "duration_prior" to "align_prior_matrix"
    - [x] Add `normalized_text` field 
    - [x] Fix minor bugs
- [x] Update FastPitch 
    - [x] Comment unnecessary code and suggest to remove it (+ need to discuss with @blisc)
    - [x] Refactor config 
    - [x] Add correct setup for text normalization and text tokenization to support new config
    - [x] Fix minor bugs
- [x] Update Mixer-TTS 
    - [x] Refactor config 
    - [x] Add correct setup for text normalization and text tokenization to support new config
    - [x] Fix minor bugs
    - [x] Add test to Jenkinsfile for Mixer-TTS
    - [x] Update documentation for Mixer-TTS and Mixer-TTS-X
    - [x] Add checkpoints from NGC Catalog to code
 - [x] Update HiFi-GAN
    - [x] Add checkpoints from NGC Catalog to code (+ need to discuss with @blisc)
- [x] Add & update notebooks
    - [x] Add `MixerTTS_Training.ipynb` notebook
    - [x] Update `Inference_ModelSelect.ipynb` notebook
    - [x] Fix bugs in `FastPitch_Finetuning.ipynb` notebook
  

Signed-off-by: Oktai Tatanov oktai.tatanov@gmail.com

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -2129,6 +2129,22 @@ pipeline {
...
           }
         }
+        stage('Mixer-TTS') {
+          steps {
+            sh 'python examples/tts/mixer_tts.py \
+            train_dataset=/home/TestData/an4_dataset/an4_train.json \
+            validation_datasets=/home/TestData/an4_dataset/an4_val.json \
+            sup_data_path=/home/TestData/an4_dataset/sup_data \
+            trainer.devices="[0]" \
+            +trainer.limit_train_batches=1 +trainer.limit_val_batches=1 trainer.max_epochs=1 \
```
Comment (blisc) [inline Jenkinsfile:2139]: Can you try using `+trainer.fast_dev_run=true` instead of this line?
Might want to do this for all TTS tests.

File: examples/tts/conf/hifigan/model/validation_ds/val_ds.yaml [hunk 2]
```diff
@@ -3,8 +3,8 @@ dataset:
   _target_: "nemo.collections.tts.data.datalayers.AudioDataset"
   manifest_filepath: ${validation_datasets}
   max_duration: null
-  min_duration: null
-  n_segments: -1
+  min_duration: 3
+  n_segments: 66048
```
Comment (blisc) [inline examples/tts/conf/hifigan/model/validation_ds/val_ds.yaml:7]: Can you explain why you made these changes?

File: examples/tts/conf/mixer-tts.yaml [hunk 3]
```diff
@@ -238,24 +217,32 @@ model:
       d_model: 1  # Disable scaling based on model dim
 
 trainer:
-  gpus: 1
+  num_nodes: 1
+  devices: 1
+  accelerator: gpu
+  strategy: ddp
   precision: 32
```
Comment (blisc) [inline examples/tts/conf/mixer-tts.yaml:224]: Same question as fp, why fp32?

File: nemo/collections/tts/models/fastpitch.py [hunk 4]
```diff
@@ -42,45 +39,66 @@
...
+#     preprocessor: Dict[Any, Any] = MISSING
+#     input_fft: Dict[Any, Any] = MISSING
+#     output_fft: Dict[Any, Any] = MISSING
+#     duration_predictor: Dict[Any, Any] = MISSING
+#     pitch_predictor: Dict[Any, Any] = MISSING
 
 
 class FastPitchModel(SpectrogramGenerator, Exportable):
-    """FastPitch Model that is used to generate mel spectrograms from text"""
+    """FastPitch Model that is used to generate mel spectrogram from text"""
```
Comment (blisc) [inline nemo/collections/tts/models/fastpitch.py:46]: Let's revert this

**Review by blisc — CHANGES_REQUESTED** [review]

File: examples/tts/conf/hifigan/model/validation_ds/val_ds.yaml (hunk 2 above)
Comment (Oktai15) [reply examples/tts/conf/hifigan/model/validation_ds/val_ds.yaml:7]: @blisc, validation step faster with this setting (instead of using `n_segments=-1`), also it can potentially produce OOM if dataset has long samples. By the way, `validation_ds/val_ds_finetune.py` has exactly this setting.

File: Jenkinsfile (hunk 1 above)
Comment (Oktai15) [reply Jenkinsfile:2139]: Ok.

File: nemo/collections/tts/models/fastpitch.py (hunk 4 above)
Comment (Oktai15) [reply nemo/collections/tts/models/fastpitch.py:46]: Ok.

File: examples/tts/conf/fastpitch_align_v1.05.yaml [hunk 5]
```diff
@@ -222,23 +219,26 @@ model:
       d_model: 1  # Disable scaling based on model dim
 
 trainer:
-  devices: -1 # number of gpus
-  max_epochs: 1500
   num_nodes: 1
+  devices: 1
   accelerator: gpu
   strategy: ddp
+  precision: 32
```
[the remark this answers is not in the document]
Comment (Oktai15) [reply examples/tts/conf/fastpitch_align_v1.05.yaml:226]: Ok.

File: examples/tts/conf/mixer-tts.yaml (hunk 3 above)
Comment (Oktai15) [reply examples/tts/conf/mixer-tts.yaml:224]: Ok.

File: tutorials/tts/FastPitch_Finetuning.ipynb [hunk 6]
```diff
@@ -149,7 +149,21 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ed1898b9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "home_path = !(echo $HOME)\n",
+    "home_path = home_path[0] \n",
+    "print(home_path)"
```
[the remark this answers is not in the document]
Comment (Oktai15) [reply tutorials/tts/FastPitch_Finetuning.ipynb:166]: Works

File: Jenkinsfile (hunk 1 above)
Comment (blisc) [reply Jenkinsfile:2139]: Reverted
